addons/mail/models/mail_message.py

```diff
@@ -1135,6 +1135,7 @@ def _message_notification_format(self):
         the message was successfully sent or if an exception or bounce occurred.
         """
         return [{
+            'author': {'id': message.author_id.id, 'type': "partner"},
```

Comment:
author_id can be undefined: it could also be guest author (other field), or be completely not set (received by email, no linked partner)